http-server-tck/src/main/java/io/micronaut/http/server/tck/tests/forms/FormsJacksonAnnotationsTest.java

```diff
@@ -0,0 +1,97 @@
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.tck.AssertionUtils;
+import io.micronaut.http.tck.HttpResponseAssertion;
+import io.micronaut.http.tck.TestScenario;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SuppressWarnings({
+        "java:S5960", // We're allowed assertions, as these are used in tests only
+        "checkstyle:MissingJavadocType",
+        "checkstyle:DesignForExtension"
+})
+public class FormsJacksonAnnotationsTest {
+    private static final String SPEC_NAME = "FormsJacksonAnnotationsTest";
+    private static final String JSON_WITH_PAGES = "{\"title\":\"Building Microservices\",\"paginas\":100}";
+    private static final String JSON_WITHOUT_PAGES = "{\"title\":\"Building Microservices\"}";
+
+    @Test
+    public void serverFormSubmissionsSupportJacksonAnnotations() throws IOException {
+        String body = "title=Building+Microservices&paginas=100";
+        assertWithBody(body, JSON_WITH_PAGES);
+
+        body = "title=Building+Microservices&pages=";
+        assertWithBody(body, JSON_WITHOUT_PAGES);
+    }
+
+    @Test
+    public void httpClientFormSubmissionsDoesNotSupportJacksonAnnotations() throws IOException {
+        Book book = new Book("Building Microservices", 100);
+        // Jackson annotations (@JsonProperty) are not supported by the HTTP Client and form-url encoded payload.
+        assertWithBody(book, JSON_WITHOUT_PAGES);
+    }
+
+    private static void assertWithBody(Object body, String expectedJson) throws IOException {
+        TestScenario.builder()
+                .specName(SPEC_NAME)
+                .request(HttpRequest.POST("/book/save", body).contentType(MediaType.APPLICATION_FORM_URLENCODED_TYPE))
+                .assertion((server, request) ->
+                        AssertionUtils.assertDoesNotThrow(server, request, HttpResponseAssertion.builder()
+                                .status(HttpStatus.OK)
+                                .assertResponse(httpResponse -> {
+                                    Optional<String> bodyOptional = httpResponse.getBody(String.class);
+                                    assertTrue(bodyOptional.isPresent());
+                                    assertEquals(expectedJson, bodyOptional.get());
+                                })
+                                .build()))
+                .run();
+    }
+
+    @Requires(property = "spec.name", value = SPEC_NAME)
+    @Controller("/book")
+    static class SaveController {
+        @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+        @Post("/save")
+        Book save(@Body Book book) {
+            return book;
+        }
+    }
+
+    @Introspected
+    record Book(@JsonProperty("title") @NonNull String title, @JsonProperty("paginas") @Nullable Integer pages) {
```

Comment:
If I use: 

```
@Introspected
record Book(@NonNull String title, @JsonProperty("paginas") @Nullable Integer pages) {}
```
 
 without the redundant ` @JsonProperty("title")` in GraalVM Native Image it renders as: 
 
`{“paginas":100}` instead of `{"title":"Building Microservices","paginas":100}`

Any idea? The TCK uses Micronaut Jackson Databind